### ISSUE TITLE
Agree on prototype scope

### ISSUE THREAD

**Issue by hoegrammer** [hoegrammer/democracy#21]
Need to ensure we all agree on what functionality we want in the prototype, what to mock up etc.

This is a blocker for three issues: Front end requirements, Back-end requirements and User Acceptance Tests.

**Comment by AaronNGray** [comment]
Basically AFAICS we need user accounts. Storage of motions, with title, proposer, date, version, and four signatories, as well as the text. Then we need voting on motions with user, time, and vote, yes, no, or abstain. The API's can be the dividing line between front and backend. We do need sign in and email verification to avoid annoying third party behaviour like spamming, etc, Just imagine if the conservative hackers got hold of the URL ! We can have email invites to allow people to share the app.

I am going to convert Drywall's templates to Underscore ones so we can use that and may possibly do away with Bootstrap or convert it to work with User-Agent info so we can get proper displays across devices which Bootstrap is meant to give but does not really.

I will draw up an initial REST/CRUD API spec that we can use as a basic starting point

Regarding the front end. I think we want single pages for login, signup, etc. And two Backbone Single Page Applications, one for motion creation, and one for voting. The main page can be a paged list of motion titles, these can be click on to be take to maybe a tabbed page with a motion details and separate tab for voting, or have the voting below, which I am not so keen on. On the main page there should be a button for creating a motion that takes you to a separate editing page with the title, creator, and text. 

Also we need options for Signing motions (the four signatories), and also ideally we need versioning of the motions. Another thing I am looking at is a forum style discourse for discussing motions.

I have worked out some very good techniques for anonymous identities, vote verification and zero knowledge Block Chain storage of the votes, but we can reserve that for the actual things as long as we have HTTPS SSL/TLS. The EFF are planning on doing free multi domain SSL certs for September. I am just hoping thats the beginning of September rather than the end :- https://letsencrypt.org/

**Comment by AaronNGray** [comment]
Prototype API document :-
https://drive.google.com/drive/u/0/folders/0B23a5KI12BkXfndfR3Zsd3pYTUFVXzZsdDktQU13SGhkV1dRNzN3RElKdjVDZW02aWlPOTQ

**Comment by hoegrammer** [comment]
I like the API doc. 

I agree with this:

"Regarding the front end. I think we want single pages for login, signup, etc. And two [screens], one for motion creation, and one for voting. The main page can be a paged list of motion titles, these can be click on to be take to maybe a tabbed page with a motion details and separate tab for voting, or have the voting below, which I am not so keen on. On the main page there should be a button for creating a motion that takes you to a separate editing page with the title, creator, and text."

I've changed "Single Page Applications" to "screens" because I think the whole thing should be a single page application. 

I'm glad to hear you have ideas for anon identities and so on but I don't think we need anything like that for the prototype. Also there's no point having SSL for the prototype as it will only be used with made-up data. The same goes for email verfication.

Four signatories, versioning and forum - these would be nice to have in the prototype but the main stuff needs to be in place first

I'm confused why you are working on templates. I thought Peter and I were doing the front end and you were doing the back end? I would rather not use Drywall, at least for the prototype, as there is no need for it and I am not convinced it's right for us anyway.

**Comment by AaronNGray** [comment]
I don't think Chris will want to use it unless it has the functionality I
have outlined. He will just use DemocracyOS as a demo otherwise.

**Comment by AaronNGray** [comment]
The templates are for login, signup, and invite pages, as well as the admin
side of the app, which drywall gives. I have been working on an looking at
this problem and working with Chris Pane, since March/April. We really need
more functionality if you look at Loomio and DemocracyOS'es level of
functionality that's what Chris will be expecting.

**Comment by AaronNGray** [comment]
If we allow motions to be created, i.e. people to enter text that will be
displayed to other Green Party members at the conference, this needs
password protection so we do not get SPAM or any other nasty things being
entered. Really we need a closed system as far as is possible.

**Comment by AaronNGray** [comment]
ALSO We cannot do anything that might upset the SOC at conference.

**Comment by AaronNGray** [comment]
r.e. SSL its going to be free and we have to look professional otherwise
people will pick on things like that. There will be people using the demo
prototype who we don't get a chance to talk to so it has to look as
professional as possible.

**Comment by hoegrammer** [comment]
I don't think there is any chance of the three of us developing and testing that in 2 months. Even with a team of experienced developers working full time it would be pushing it.

**Comment by AaronNGray** [comment]
I don't actually see that as being much extra work, most of the
functionality is there in drywall.
But it would still not demonstrate as much functionality as DemocracyOS or
Loomio.

I think Chris will still think it is probably better to use DemocracyOS and
Loomio as demonstrations for now then as they have been deployed and
had proper user testing for a reasonable period of time.

**Comment by hoegrammer** [comment]
It might be worth picking one and working on a few improvements with the time we have

**Comment by AaronNGray** [comment]
That would mean hosting it ourselves. I went to the trouble of hosting
DemocracyOS with SSL Certificate for Chris for a hacker meeting in April
this year and he did not use it preferring to use DemocracyOS'es own
demostarion.
